### PR TITLE
[HDRP] Improve path traced subsurface scattering for transmissive surfaces

### PR DESCRIPTION
**Pull request by pmavridis**
### Purpose of this PR
https://fogbugz.unity3d.com/f/cases/1329403/

There was a problem when computing subsurface scattering for transmissive thin/open surfaces

This is what the rasterized scene looks like:
image

Path tracer BEFORE this PR:
image

Path tracer AFTER this PR:
image

(The text is missing from the path traced images because the emitted UI geometry is not part of the acc structure)

---
### Testing status
Tested using the scene that was attached to the bug ticket, as you see above

### CODE REVIEW

**Review by RemyUnity — APPROVED** [review]
LGTM, thanks for the comparison screenshots 🟢

**Review by eturquin — APPROVED** [review]

**Comment by remi-chapelain** [comment]
How about adding this as a test scene in the DXR Test project ?

**Comment by pmavridis** [comment]
> How about adding this as a test scene in the DXR Test project ?

That's a good idea, I will followup with another PR, after I finish with some other work.

**Comment by remi-chapelain** [comment]
> That's a good idea, I will followup with another PR, after I finish with some other work.

Thanks 🙏

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl [hunk 1]
```diff
@@ -607,6 +607,16 @@ bool RandomWalk(float3 position, float3 normal, float3 diffuseColor, float3 mean
         result.exitNormal = intersection.value;
     }
 
+// if the material is transmissive, pick a side for the normal randomly to properly handle thin surfaces (case 1329403)
+#ifdef _MATERIAL_FEATURE_TRANSMISSION
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/PathTracing/Shaders/PathTracingBSDF.hlsl:611]: I will accept the PR for the fix, but let's have in mind that _MATERIAL_FEATURE_TRANSMISSION isn't a standard name for "transmission". This is define only per Material and Material don't have obligation to reuse this name (on raster side). We may need to revisit how we handle transmission from a general point of view later.

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
As mention by Remi, please add a test in another PR

**Pull request merged**